Local Jnet - heap space

as many who use win7 have many troubles like that
        at clojure.main$null_opt.invoke(main.clj:342)                               
        at clojure.main$main.invokeStatic(main.clj:424)                             
        at clojure.main$main.doInvoke(main.clj:387)                                 
        at clojure.lang.RestFn.applyTo(RestFn.java:137)                             
        at clojure.lang.Var.applyTo(Var.java:702)                                   
        at clojure.main.main(main.java:37)                                          
Caused by: java.lang.OutOfMemoryError: Java heap space                              
        at java.nio.HeapByteBuffer.<init>(HeapByteBuffer.java:57)                   
        at java.nio.ByteBuffer.allocate(ByteBuffer.java:335)                        
        at org.httpkit.client.HttpsRequest.<init>(HttpsRequest.java:25)             
        at org.httpkit.client.HttpClient.exec(HttpClient.java:307)                  
        at org.httpkit.client$request.invokeStatic(client.clj:195)                  
        at org.httpkit.client$request.doInvoke(client.clj:99)                       
        at clojure.lang.RestFn.invoke(RestFn.java:423)                              
        at org.httpkit.client$get.invokeStatic(client.clj:208)                      
        at org.httpkit.client$get.doInvoke(client.clj:208)                          
        at clojure.lang.RestFn.invoke(RestFn.java:423)                              
        at tasks.nrdb$download_card_image.invokeStatic(nrdb.clj:211)                
        at tasks.nrdb$download_card_image.invoke(nrdb.clj:207)                      
        at clojure.lang.AFn.applyToHelper(AFn.java:156)                             
        at clojure.lang.AFn.applyTo(AFn.java:144)                                   
        at clojure.core$apply.invokeStatic(core.clj:657)                            
        at clojure.core$apply.invoke(core.clj:652)                                  
        at throttler.core$fn_throttler$fn__603$fn__604.doInvoke(core.clj:171)       
        at clojure.lang.RestFn.invoke(RestFn.java:421)                              
        at clojure.core.protocols$fn__7852.invokeStatic(protocols.clj:168)          
        at clojure.core.protocols$fn__7852.invoke(protocols.clj:124)                
        at clojure.core.protocols$fn__7807$G__7802__7816.invoke(protocols.clj:19)   
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)         
when try to build local jnet its become half empty space

--- a/src/clj/tasks/nrdb.clj
+++ b/src/clj/tasks/nrdb.clj
@@ -14,7 +14,7 @@
 (declare faction-map)
 
 (def ^:const base-url "http://www.netrunnerdb.com/api/2.0/public/")
-(def ^:const base-image-url "http://www.netrunnerdb.com/card_image/")
+(def ^:const base-image-url "https://www.cardgamedb.com/forums/uploads/an/")
 
 (defmacro rename
   "Rename a card field"
@@ -38,6 +38,7 @@
    :size (fn [[k v]] [:bigbox (> (or v -1) 20)])
    :code identity
    :position identity
+   :ffg_id identity
    })
 
 (def mwl-fields
@@ -185,12 +186,17 @@
               acc))
           c fields))
 
+(defn- make-image-url
+  "Create a URI to the card in CardGameDB"
+  [card set]
+  (str base-image-url "med_ADN" (:ffg_id set) "_" (:number card) ".png"))
+
 (defn- get-uri
   "Figure out the card art image uri"
-  [card]
+  [card set]
   (if (contains? card :image_url)
     (:image_url card)
-    (str base-image-url (:code card) ".png")))
+    (make-image-url card set)))
 
 (defn- add-card-fields
   "Add additional fields to the card documents"
@@ -201,7 +207,7 @@
       (assoc :setname (:name s)
              :cycle_code (:cycle_code s)
              :rotated (:rotated s)
-             :image_url (get-uri c)
+             :image_url (get-uri c s)
              :normalizedtitle (string/lower-case (deaccent (:title c)))))))
 
 (defn fetch-data

--- a/src/clj/tasks/nrdb.clj
+++ b/src/clj/tasks/nrdb.clj
@@ -14,7 +14,8 @@
 (declare faction-map)
 
 (def ^:const base-url "http://www.netrunnerdb.com/api/2.0/public/")
-(def ^:const base-image-url "https://www.cardgamedb.com/forums/uploads/an/")
+(def ^:const cgdb-image-url "https://www.cardgamedb.com/forums/uploads/an/")
+(def ^:const nrdb-image-url "https://netrunnerdb.com/card_image/")
 
 (defmacro rename
   "Rename a card field"
@@ -189,7 +190,9 @@
 (defn- make-image-url
   "Create a URI to the card in CardGameDB"
   [card set]
-  (str base-image-url "med_ADN" (:ffg_id set) "_" (:number card) ".png"))
+  (if (:ffg_id set)
+    (str cgdb-image-url "med_ADN" (:ffg_id set) "_" (:number card) ".png")
+    (str nrdb-image-url (:code card) ".png")))
 
 (defn- get-uri
   "Figure out the card art image uri"
@@ -234,9 +237,15 @@
 
 (defn- download-card-image
   "Download a single card image from NRDB"
-  [acc card]
-  (println "Downloading: " (:title card))
-  (concat acc (list [card (http/get (:image_url card) {:as :byte-array :timeout 120000})])))
+  [card]
+  (println "Downloading: " (:title card) "\t\t(" (:image_url card) ")")
+  (http/get (:image_url card) {:as :byte-array :timeout 120000}
+            (fn [{:keys [status body error]}]
+              (case status
+                404 (println "No image for card" (:code card) (:title card))
+                200 (with-open [w (io/output-stream (.getPath (card-image-file card)))]
+                      (.write w body))
+                (println "Error downloading art for card" (:code card) error)))))
 
 (def download-card-image-throttled (throttle-fn download-card-image 5 :second))
 
@@ -252,16 +261,11 @@
            missing (count missing-cards)]
        (when (> missing 0)
          (println "Downloading art for" missing "cards...")
-         (let [futures (reduce download-card-image-throttled nil missing-cards)]
-           (doseq [[card resp] futures]
-             (let [status (:status @resp)]
-               (cond
-                 (= 404 status) (println "No image for card" (:code card) (:title card))
-                 (= 200 status) (with-open [w (io/output-stream (.getPath (card-image-file card)))]
-                                  (.write w (:body @resp))
-                                  (println "Downloaded art for card" (:code card) (:title card)))
-                 :else (println "Error downloading art for card" (:code card) (:error @resp)))))
-           (println "Finished downloading card art"))))))
+         (let [futures (doall (map download-card-image-throttled missing-cards))]
+           (doseq [resp futures]
+             ; wait for all the GETs to complete
+             (:status @resp)))
+         (println "Finished downloading card art")))))
   
 (defn fetch-cards
   "Find the NRDB card json files and import them."

